### DISCUSSION TITLE
An idle state when `enabled` flag is falsy & cached data exists

### DISCUSSION THREAD

**Discussion by jxom** [TanStack/query#3405] [Ideas] [answered]
Hello!

Under the **Disabling/Pausing Queries** section of the docs, it states that when the `enabled` flag is falsy & **cached data exists**, then the query will be initialized in a `success` state. 

If `enabled` is falsy & cached data exists, is it possible for the query result to initialize in an `idle` state, and then when `refetch` is invoked, it will immediately resolve to a `success` state if there is cached data?

**Comment by TkDodo** [comment, marked as answer]
> If enabled is falsy & cached data exists, is it possible for the query result to initialize in an idle state, and then when refetch is invoked, it will immediately resolve to a success state if there is cached data?

no, it's not. The `state` gives information about `data`. `loading` means there is no data yet, as does `idle`. `success` means you have `data`. You can also be fetching while you are in success state if a background refetch happens.

FYI, in v4, the idle state will be removed, see: https://react-query-alpha.tanstack.com/guides/queries
